keps/sig-storage/1432-volume-health-monitor/README.md

```diff
@@ -674,8 +679,193 @@ _This section must be completed when targeting alpha to a release._
+  status code is `OK`, the call is successful; otherwise, it is not successful. We
+  can look at the ratio of successful vs non-successful statue codes to figure out
+  the success/failure ratio.
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+  Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+  implementation difficulties, etc.).
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Think about both cluster-level services (e.g. metrics-server) as well
+  as node-level agents (e.g. specific version of CRI). Focus on external or
+  optional services that are needed. For example, if this feature depends on
+  a cloud provider API, or upon an external software-defined storage or network
+  control plane.
+
+  For each of these, fill in the following—thinking about running existing user workloads
+  and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]: installation of csi-external-health-monitor-controller and csi-external-health-monitor-agent sidecars
+    - Usage description:
+      - Impact of its outage on the feature: Installation of csi-external-health-monitor-controller and csi-external-health-monitor-agent sidecars are required for the feature to work. If csi-external-health-monitor-controller is not installed, abnormal volume conditions will not be reported as events on PVCs. Similarly, if csi-external-health-monitor-agent is not installed, abnormal volume conditions will not be reported as events on Pods.
+      - Impact of its degraded performance or high-error rates on the feature: If abnormal volume conditions are reported with degraded performance or high-error rates, that would affect how soon or how accurately users could manually react to these conditions.
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  Describe them, providing:
+  - API call type (e.g. PATCH pods): Only events will be reported to PVCs or Pods if this feature is enabled.
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+  focusing mostly on:
+  - components listing and/or watching resources they didn't before
+    csi-external-health-monitor-controller and csi-external-health-monitor-agent sidecars.
+    There is a monitor interval for the controller and one for the agent to control how often
+    to check the volume health. It is configurable with 1 minute as default.
```

Comment:
> We have a monitor interval to control how often to check the volume health. It is configurable with 1 minute as default.
> I'm not aware there is a "similar event collapsing" feature on the event broadcaster. Will take a look.

We found that when scaled out across many nodes, low frequency checks can still produce high volumes of events.  To control this, we created the eventrecorder and have options on it to control QPS per key.  Can you have a look and be sure that you're collapsing keys and have a slow update cadence per key?  I think kubelets have a modest burst and then do something like one event every five minutes per key to avoid overloading kube-apiserver.

Reply:
Sure.  Added to the KEP that we should use this "similar event collapsing" feature and also consider changing the default from 1 to 5 minutes.